### PR TITLE
Soft deprecation for Datasets

### LINKED ISSUE #1426: add soft deprecation path for removed datasets

### PR DESCRIPTION
**Pull request by CarloLucibello**
Add a soft deprecations path for the datasets I brutally removed in #1377 .
I added the old tests back, verified they passed, then removed them again. 

Fix #1426

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
Let's have the tests in still, they can be removed when we remove the code.

File: src/data/Data.jl [hunk 1]
```diff
@@ -7,32 +7,56 @@ include("dataloader.jl")
...
+    end
+    mv(tmppath, path; force=true)
 end
-module Sentiment
-train() = error("Dataset is deprecated, use MLDatasets.jl instead.")
-test() = error("Dataset is deprecated, use MLDatasets.jl instead.")
-dev() = error("Dataset is deprecated, use MLDatasets.jl instead.")
+
+function __init__()
+  mkpath(deps())
```
Comment (DhairyaLGandhi) [inline src/data/Data.jl:37]: Is this necessary? It breaks package compiler with flux

Comment (CarloLucibello) [reply src/data/Data.jl:37]: I don't know, it was already there

Comment (CarloLucibello) [reply src/data/Data.jl:37]: I don't why it breaks package compiler but  I don't think we can avoid creating the directories

**Comment by CarloLucibello** [comment]
@DhairyaLGandhi bump

**Comment by CarloLucibello** [comment]
@DhairyaLGandhi can you approve here?

**Comment by CarloLucibello** [comment]
wait, the failing doctests point at something wrong

**Comment by CarloLucibello** [comment]
@DhairyaLGandhi do you understand the error in the doctests? I don't

**Comment by DhairyaLGandhi** [comment]
Yeah, hadn't approved because of failing ci

**Comment by DhairyaLGandhi** [comment]
Something changed in Iris? Will check in some time

**Comment by darsnack** [comment]
@CarloLucibello my only guess is that it's the warning statement that bypasses the `;`. I checked out your branch and ran the doctest on my machine, and the output is nearly identical:
```julia
julia> using Flux
[ Info: Precompiling Flux [587475ba-b771-5e3f-ad9e-33799f191a9c]

julia> Flux.Data.Iris.load();
[ Info: Downloading iris dataset.

julia> labels = Flux.Data.Iris.labels();
┌ Warning: Flux's datasets are deprecated, please use the package MLDatasets.jl
└ @ Flux.Data ~/.julia/packages/Flux/hTG79/src/data/Data.jl:17

julia> summary(labels)
"150-element Array{String,1}"

julia> labels[1]
"Iris-setosa"
```

Kind of makes sense for the behavior we are seeing, since it seems like the test line interpretation is shifted. I would also try adding a blank line between REPL output and the next input (just in case, not sure this is required for Documenter.jl).

**Comment by CarloLucibello** [comment]
I just removed the doctests

**Review by darsnack — DISMISSED** [review]

**Review by DhairyaLGandhi — APPROVED** [review]

**Comment by DhairyaLGandhi** [comment]
bors try

**Comment by DhairyaLGandhi** [comment]
bors try

**Comment by DhairyaLGandhi** [comment]
bors try-

**Comment by DhairyaLGandhi** [comment]
bors try

**Review by DhairyaLGandhi — APPROVED** [review]

**Pull request merged**